modules/openapi-generator/src/main/resources/C-libcurl/api-body.mustache

```diff
@@ -403,13 +418,28 @@ end:
     {{#queryParams}}
     {{^isListContainer}}
     {{#isString}}
-    free(keyQuery_{{{paramName}}});
-    free(valueQuery_{{{paramName}}});
-    keyValuePair_free(keyPairQuery_{{{paramName}}});
+    if(keyQuery_{{{paramName}}}){
+        free(keyQuery_{{{paramName}}});
+        keyQuery_{{{paramName}}} = NULL;
+    }
+    if(valueQuery_{{{paramName}}}){
+        free(valueQuery_{{{paramName}}});
+        valueQuery_{{{paramName}}} = NULL;
+    }
+    if(keyPairQuery_{{{paramName}}}){
+        keyValuePair_free(keyPairQuery_{{{paramName}}});
+        keyPairQuery_{{{paramName}}} = NULL;
+    }
     {{/isString}}
     {{#isString}}
-    free(keyQuery_{{{paramName}}});
-    keyValuePair_free(keyPairQuery_{{{paramName}}});
+    if(){
```

Comment:
Are these empty conditions expected?

Reply:
Sorry, It's my mistake.

I have fixed.

Reply:
That's ok. Looks good now.